src/utils/events.ts

```diff
@@ -0,0 +1,50 @@
+﻿/* eslint-disable @typescript-eslint/no-explicit-any */
+interface Event {
+    type: string;
+}
+
+type callback = (e: Event, ...args: any[]) => void;
+
+function getCallbacks(obj: any, type: string): callback[] {
+    if (!obj) {
+        throw new Error('obj cannot be null!');
+    }
+
+    obj._callbacks = obj._callbacks || {};
```

Comment:
You might want to use a symbol to add data on objects to avoid conflicts with json serializing / spread operators etc. etc.

Reply:
Well that was a fun dive into symbols. I really don't think I have used them before.

Unfortunately we have to keep the property name the same here since events that are triggered by the apiclient on the apiclient or connection manager still need to be handled by listeners attached in web.